### ISSUE TITLE
HMC Gibbs Hierarchical Prior Issue

### ISSUE THREAD

**Issue by ross-h1** [pyro-ppl/numpyro#1299]
I'm trying to fit a hierarchical model with HMC Gibbs. Much simplified model in the attached notebook. Works fine with a fixed hyperparameter for tau, but not at all with a prior over tau sampled in the model function. I've seen this behaviour in several models and trying to get to the bottom of it. Quite possible that I am missing something... but seems as though this should work? 

[hmc gibbs hierarchical.ipynb.zip](https://github.com/pyro-ppl/numpyro/files/7916630/hmc.gibbs.hierarchical.ipynb.zip)

**Comment by martinjankowiak** [comment]
@ross-h1 can you please post your code to a github (gist?) or the like?

**Comment by ross-h1** [comment]
yes, happy to but unless I am mistaken I think this might be described as a bug report...

**Comment by fehiepsi** [comment]
+1 for uploading the notebook to https://gist.github.com/  Then we can use https://nbviewer.org/ to view it.

**Comment by ross-h1** [comment]
et voila...

https://gist.github.com/ross-h1/677e6fb13a806bade94e387a5dda6d9e

Thanks for looking at this guys...

**Comment by fehiepsi** [comment]
Here is the rendered version of the notebook: [link](https://nbviewer.org/gist/ross-h1/677e6fb13a806bade94e387a5dda6d9e).

With your model, `p(tau, sigma | alpha, liks) = p(tau) p(sigma | alpha, liks)` so its samples using HMCGibbs should follow `dist.HalfNormal(1)` distribution (which is also its prior).

**Comment by martinjankowiak** [comment]
@ross-h1 shouldn't the posterior conditional over `alpha` be a multivariate normal distribution?

**Comment by fehiepsi** [comment]
Closing. @ross-h1, please feel free to open discussions in https://forum.pyro.ai/ .

**Comment by ross-h1** [comment]
Yep thanks for pointing out. Found another way!
